knative-kubecon/README.md

```diff
@@ -230,4 +230,132 @@ $ curl -H "Host: helloworld-openshift.myproject.example.com" "http://$(minishift
+```
+Here we can see that we've got a channel named `testchannel` and it is an `in-memory-channel` which is what we're going to use for this demo - in production we would probably use Apache Kafka.
+Let's deploy that so that we can use it in a later stage.
+
+```bash
+$ oc apply -f 010-channel.yaml -n myproject
+```
+
+Next let's take a look at the _EventSource_.
+
+```yaml
+$ less 020-k8s-event-source.yaml
+apiVersion: sources.eventing.knative.dev/v1alpha1
+kind: KubernetesEventSource
+metadata:
+  name: testevents
+spec:
+  namespace: myproject
+  serviceAccountName: default
+  sink:
+    apiVersion: eventing.knative.dev/v1alpha1
+    kind: Channel
+    name: testchannel
+```
+
+This is starting to get a little bit more interesting.
+This _EventSource_ is of type `KubernetesEventSource`.
+The `KubernetesEventSource` is available as an example _EventSource_ from the Knative project.
+It is going to receive Kubernetes platform events from a particular namespace, in this case `myproject`.
+All of the messages received from Kubernetes are going to be routed to the `sink` that is specified as part of the _EventSource_.
+Here we can see that the `testchannel` we defined earlier is specified as the `sink`.
+
+If we apply that we will see a pod created which is an instance of the `KubernetesEventSource` that is configured with the _Channel_ we defined.
+
+```bash
+$ oc apply -f 020-k8s-event-source.yaml -n myproject
+```
+
+```bash
+$ oc get pods
+NAME                                               READY     STATUS    RESTARTS   AGE
+testevents-dbclc-b6gv8-db66b4985-lqljv             2/2       Running   0          15s
+```
+
+Question: do you want to show the logs here?
+
+The EventSource is up and running and the final piece of the Eventing platform is how we wire everything together.
+This is done via a _Subscription_.
+
+```yaml
+$ less 030-subscription.yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: Subscription
+metadata:
+  name: testevents-subscription
+  namespace: myproject
+spec:
+  channel:
+    apiVersion: eventing.knative.dev/v1alpha1
+    kind: Channel
+    name: testchannel
+  subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1alpha1
+      kind: Service
+      name: helloworld-openshift
+```
+
+This _Subscription_ again references the `testchannel` and also defines a `Subscriber`, in this case the Serving application `helloworld-openshift` we built earlier.
+This basically means that we have subscribed the previously built application to the `testchannel`.
+Once we apply the `Subscription` any Kubernetes platform events from the `myproject` namespace will be routed to the `helloworld-openshift` application.
+Behind the scenes there is a `SubscriptionController` which is doing the wiring for us.
+
+```bash
+$ oc apply -f 030-subscription.yaml -n myproject
+```
+
+If we take a look at the logs of the function we will start to see some messages but may need to generate some load before anything appears.
```

Comment:
Let's not talk about "load" here since that's overloaded with generating load to the application itself.

Let's say we need to generate some events and to do that we're going to create a pod which will emit a couple of events (like pod created, container started). Makes it more explicit imho.